MDN Community: Discussion guidelines (#32620)
mdn/content: files/en-us/mdn/community/discussions/Resolving_discussions/index.md

---

> title: Resolving discussions

Author: Changed to "Managing and resolving discussions".

slug: MDN/Community/Discussions/Resolving_discussions
page-type: mdn-community-guide
---

{{MDNSidebar}}

> The MDN community is encourage to [initiate, engage in, and resolve discussions](/en-US/docs/MDN/Community/Discussions) related to the MDN Web Docs documentation.

Reviewer: ```suggestion
The MDN community is encouraged to [initiate, engage in, and resolve discussions](/en-US/docs/MDN/Community/Discussions) related to the MDN Web Docs documentation.
```

While many discussions reach broad agreement very quickly, others may stall if there is no clear path to resolution: for example due to conflicting opinions. This process explains how you can work with the community in order to progress discussions through to resolution in a reasonable timeframe.

## Moving a discussion to a resolution

Most discussions do not need a formal resolution process. These MDN discussion guidelines are here for the discussions that aren't moving forward toward a conclusion or would otherwise benefit from a formal process:

> 1. Each discussion is held / rooted in a [discussion on Github](https://github.com/orgs/mdn/discussions), which is the "source of truth" for the topic.

Reviewer 2: We don't need to say this, assuming it is covered on the parent page https://developer.mozilla.org/en-US/docs/MDN/Community/Discussions

Author: We do need to say #1, because we need to be clear that meeting minutes, verbal discussions, issues, and PRs are not sources. Recaps of any meetings, discussions, issues, and PRs must be documented in the discussion as the "source of truth". Also, we don't know if someone is going re-read the top level page when following the guidelines on this page.

Reviewer 2: Got it! In that case, might help to add this clarification in.

> 2. Each topic of discussion needs a driver. The driver is responsible for:

Reviewer 2: This does not need to be a numbered step
```suggestion
Each topic of discussion needs a driver. The driver is responsible for:
```

Author: we have a 3 and a 1 so keeping the 2.

Reviewer 2: Yes, I think bulleted makes better sense

   - Guiding the conversation.
   - Making people aware the discussion exists.
   - Setting the timeline, informing people of it, changing the timeline as necessary, and sticking to the timeline as appropriate.

>    - Informing all the appropriate channels - Slack, Discord, tagging people Github, and other channels if appropriate - of needed feedback by specific dates.

Reviewer: ```suggestion
   - Informing all the appropriate channels - Slack, Discord, tagging people on GitHub, and other channels if appropriate - of needed feedback by specific dates.
```

   - Bring up the topic in community and weekly meetings.
   - Organizing a synchronous, online face-to-face meeting if required (if there is disagreement). Face-to-face meetings should be rare and only if required (See #3).

>    - Summarizing face-to-face conclusions in the relevant discussion on https://github.com/orgs/mdn/discussions within the relevant discussion.

Reviewer 3: ```suggestion
   - Summarizing face-to-face conclusions in the relevant discussion on [Discussions](https://github.com/orgs/mdn/discussions).
```

   - Guiding the implementation of the discussion results or working with the appropriate team lead to ensure that the results are implemented.

3. Face-to-face meetings regarding a discussion should only be called if there is disagreement.

   - Face-to-face meetings must be announced in all relevant [communication channels](/en-US/docs/MDN/Community/Communication_channels), such as Slack, Discord, GitHub discussion, etc.
   - Conclusions of each face-to-face meeting must be entered into the GitHub discussion, which remains as the source of truth for the discussion.
   - The driver is responsible for calling the meeting, informing all parties, reporting results back into the GitHub discussion.

Once agreement has been achieved, the discussion can be closed, the resolution announced, and the plan for implementing the resolution can be put it into action!

> ## Discussion timeline

Reviewer 2: Starting the document with this section might be a good idea. That is, we first define how to determine if a discussion is stalled and establish the expected timeline for resolution. And then adopt other strategies when we see there is no progress
```suggestion
## Expected turnaround times for discussions
```

> Every discussion will follow a different timeline depending on the complexity of the topic, the levels of agreement and disagreement, and the parties involved. Most discussions should be resolved in two months, allowing for the topic to be addressed at two monthly and / or nine weekly meetings, while ensuring that people with differing views have to opportunity to hear about and contribute to the discussion.

Reviewer 2: - What do you think about removing "parties involved"? It is ultimately about the consensus.
```suggestion
Every discussion will have a different timeline depending on the complexity of the topic and the consensus levels. Ideally, most discussions should be resolved within two months, allowing for the topic to be addressed at various internal meetings. This timeframe ensures diverse viewpoints are considered and everyone interested has the opportunity to contribute to the discussion.
```

1. Post the discussion and assign a driver.
2. Identify any key stakeholders and needed approvers (the individuals who need to weigh in on the topic and give their approval), if any.
3. Inform the approvers and other essential voices of the discussion and your proposed timeline. If needed, reach out to them again after 2 weeks, and weekly thereafter, until they provide feedback.

> 4. Add the discussion topic to the agendas of relevant meetings.

Reviewer 2: We'll not know beforehand the discussions that will get stalled. Do these steps need to be done for every discussion opened? Folks in the community will not know who the drivers, stakeholders, and approvers are.

Author: the intro explains when this guide, and therefore these steps , are relevant

---
title: Resolving discussions
slug: MDN/Community/Discussions/Resolving_discussions
page-type: mdn-community-guide
---

{{MDNSidebar}}

> MDN Web Docs is an open source project. The community is encourage to initiate, engage in, and resolve discussions related to the MDN Web Docs documentation.
> 
> At times, people will have differing views. If a discussion is controversial or otherwise requires a resolution before action is taken (or to decide to not take action), community members are encouraged to seek input and work together with interested parties toward a resolution.
> 
> If you are proposing new content or features, please open a new issue in the [relevant MDN GitHub repository](https://github.com/mdn/), such as [MDN Content](https://github.com/mdn/content/issues/) or [Yari](https://github.com/mdn/yari/issues/). If, however, you are suggesting changes to processes or other changes that would benefit from or need input from our partners and the wider community, please read the [discussion guidelines](/en-US/docs/MDN/Community/Discussions), [start a discussion](https://github.com/orgs/mdn/discussions), and drive the discussion to a workable and supported resolution. The steps below provide guidance to achieve a resolution with maximum community input in a reasonable amount of time.

Reviewer 4: This is too much context/feels a bit repetitive. This is what I think is important:

```suggestion
The MDN community is encourage to [initiate, engage in, and resolve discussions](https://developer.mozilla.org/en-US/docs/MDN/Community/Discussions) related to the MDN Web Docs documentation.

While many discussions reach broad agreement very quickly, others may stall if there is no clear path to resolution: for example due to conflicting opinions.
This process explains how you can work with the community in order to progress discussions through to resolution in a reasonable timeframe.
```

## Moving a discussion to a resolution

Most discussions do not need a formal resolution process. These MDN discussion guidelines are here for the discussions that aren't moving forward toward a conclusion or would otherwise benefit from a formal process:

1. Each discussion is held / rooted in a [discussion on Github](https://github.com/orgs/mdn/discussions), which is the "source of truth" for the topic.

> 2. Each topic of discussion needs a driver. The driver is responsible for:

Reviewer 4: I prefer the term owner to driver.

Author: i prefer the term too, but the driver may not be the owner. The person driving the discussion (the driver) to a resolution may not be the same person who opened the discussion (the owner).

Reviewer 2: When a discussion is initiated by maintainers, is it okay to go with the assumption that the author is the driver? How will community determine a driver? Also, the question of selecting a driver would arise only if a discussion hasn't progressed.

In the timeline section, we need to define the period of inactivity or non-resolution beyond which a discussion can be considered as stalled.

Author: That is why this was originally called "resolving". Discussions only need drivers, guidelines, and timelines if they need help being resolved. This is expressed in the intro. 

Generally, the initiator may be the driver, but sometimes people start discussions but aren't invested in the outcome, while someone else may be more passionate about it. I want to provide guidelines open to interpretation, not rules... but someone might be more vested than me in this discussion ;)

Reviewer 4: > The person driving the discussion (the driver) to a resolution may not be the same person who opened the discussion (the owner).

The person who cares is the owner, whoever posted it :-). The author is likely to be the owner.

But that's just semantics. The point here is that it is possible to look at this text and say "you mean the owner, or the author right". So perhaps a definition is needed. 

How does the driver get picked - it's obvious who the author is?

Author: The driver will generally be the author and owner, but doesn't have to be. Defining it to that person limits it. Leaving it undefined allows for anyone with the energy to move the discussion forward to take on that responsibility.

 I guess we could just write what I just said if required.

   - Guiding the conversation.
   - Making people aware the discussion exists.
   - Setting the timeline, informing people of it, changing the timeline as necessary, and sticking to the timeline as appropriate.
   - Informing all the appropriate channels - slack, discord, tagging people Github, and other channels if appropriate - of needed feedback by specific dates.
   - Bring up the topic in community and weekly meetings.
   - Organizing a synchronous, online face-to-face meeting if required (if there is disagreement). Face-to-face meetings should be rare and only if required (See #3).
   - Summarizing face-to-face conclusions in the relevant discussion on https://github.com/orgs/mdn/discussions within the relevant discussion.
   - Guiding the implementation of the discussion results or working with the appropriate team lead to ensure that the results are implemented.

3. Face-to-face meetings regarding a discussion should only be called if there is disagreement.

>    - Face-to-face meetings must be announced in all [communication channels](/en-US/docs/MDN/Community/Communication_channels) (Slack, Discord, GitHub discussion, etc.).

Reviewer 4: I will never go on discord so I will never engage in this process. In any case, I think we have too many channels. Minimally community call and slack.

Author: made this a bit more "appropriate channels" over "required channels" as not everyone is on discord or slack. Some may be more webDX community group based or similar.

   - Conclusions of each face-to-face meeting must be entered into the GitHub discussion, which remains as the source of truth for the discussion.
   - The driver is responsible for calling the meeting, informing all parties, reporting results back into the GitHub discussion.

Once agreement has been achieved, the discussion can be closed, the resolution announced, and the plan for implementing the resolution can be put it into action!

## Discussion timeline

Every discussion will follow a different timeline depending on the complexity of the topic, the levels of agreement and disagreement, and the parties involved. Most discussions should be resolved in two months, allowing for the topic to be addressed at two monthly and / or nine weekly meetings, while ensuring that people with differing views have to opportunity to hear about and contribute to the discussion.

1. Post the discussion and assign a driver.
2. Identify any essential voices and needed approvers (the individuals who need to weigh in on the topic and give their approval), if any.

> 3. Inform the approves and other essential voices of the discussion and your proposed timeline. If needed, reach out to them again after 2 weeks, and weekly thereafter, until they provide feedback.

Reviewer 4: ```suggestion
3. Inform the approvers and other essential voices of the discussion and your proposed timeline. If needed, reach out to them again after 2 weeks, and weekly thereafter, until they provide feedback.
```

4. Add the discussion topic to the agendas of relevant meetings.
5. After a month, sort through the feedback, discussions, and agreements, and formalate an initial plan merging the feedback into a possible action plan. Re-inform and re-request feedback from all interested parties, including everyone who participated in the discussion in any way.
6. During the second month, keeping reaching out the the community for feedback on the proposed plan, making updates to the plan in light of any new feedback. Rinse. Repeat.

> 7. If there are points of contention, arrange an online, synchronous, face-to-face meeting to resolve any remaining disagreements on in the discussion thread.

Reviewer 4: ```suggestion
7. If there are points of contention, arrange an online, synchronous, face-to-face meeting to resolve any remaining disagreements (as captured in the discussion thread).
```

8. Keep the discussion threads alive during the second month as you work with the community towards the resolution.
9. [Create an issue](/en-US/docs/MDN/Community/Issues) with the final plan for implementing the resolution and put it into action.
10. Close the discussion.

> If the discussion involves reaching out to and receiving feedback from experts, the timeline above can be extended as needed.

Reviewer 4: Might be nice to add a checklist section at the end that a discussion owner can copy/paste in and then checkoff as each step is complete.
